scripts/scripts.js

```diff
@@ -121,6 +121,78 @@ function decorateVideos(el) {
   });
 }
 
+function handlePost(response, message) {
+  if (!response.ok) {
+    message.innerText = 'Invitation Failed. Try Again.';
+  } else {
+    message.innerText = 'Invitation Successful!';
+  }
+}
+
+function submitInvite(event) {
+  // validate fields
+  let valid = 0;
+  const messageDiv = event.target.parentNode.parentNode.querySelector('.row.message');
+  const name = event.target.parentNode.parentNode.querySelector('.row > input[id=name]').value;
+  const email = event.target.parentNode.parentNode.querySelector('.row > input[id=email]').value;
+  const validRegex = /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+
+  if (name === '') {
+    messageDiv.innerText = 'Name is required';
+    valid += 1;
+  }
+  if (!email.match(validRegex)) {
+    messageDiv.innerText = 'Valid email is required';
+    valid += 1;
+  }
+  if (valid === 0) {
+    fetch('/invite-form', {
+      method: 'POST',
+      cache: 'no-cache',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        data: {
+          name,
+          email,
+        },
+      }),
+    }).then((response) => handlePost(response, messageDiv));
+  }
+}
+
+function hideModal(event) {
+  if (event.target.matches('.invite-modal') || event.target.matches('.close-invite')) {
+    const modal = document.querySelector('.invite-modal');
+    modal.style.display = 'none';
+    modal.removeEventListener('click', hideModal);
+  }
+}
+
+function showModal() {
+  const modal = document.querySelector('.invite-modal');
+  modal.style.display = 'block';
+  modal.addEventListener('click', (event) => {
+    hideModal(event);
+  });
+  modal.querySelector('.row > input[type=button]').addEventListener('click', submitInvite);
+  modal.querySelector('div > .close-invite').addEventListener('click', (event) => {
+    hideModal(event);
+  });
+}
+
+function makeInviteModal() {
+  const modal = document.querySelector("[title='Share with a friend or colleague.']");
+  if (modal) {
+    modal.addEventListener('click', () => {
+      showModal();
+    });
+    // eslint-disable-next-line no-script-url
+    modal.href = 'javascript:void(0);';
```

Comment:
Any particular reason to `void` the `href` here? You could do `ev.preventDefault()` in the handler to forcibly disable the default behavior instead

Reply:
No reason. Going old school. I'll change it.